Detour/Include/DetourCommon.h

```diff
@@ -482,6 +483,15 @@ inline void dtSwapEndian(float* v)
 void dtRandomPointInConvexPoly(const float* pts, const int npts, float* areas,
 							   const float s, const float t, float* out);
 
+template<typename TypeToRetrieveAs, typename BufferType>
+TypeToRetrieveAs* getThenAdvanceBufferPointer(BufferType*& buffer, const size_t distanceToAdvance)
```

Comment:
Good point - I'll prefix with `dt`.

`BufferType` doesn't strictly need to be a template argument but if I leave the parameter as `void*&` then every call site needs to cast to `void*`, which is quite cumbersome - the template just helps with readability.